scipy/stats/_morestats.py

```diff
@@ -3890,3 +3890,68 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate', *,
     if not normalize:
         res *= (high-low)/(2.*pi)  # [1] (2.3.14) w/ (2.3.7)
     return res
+
+#@_axis_nan_policy_factory(
+#    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1, default_axis=None
+#)
```

Comment:
```suggestion
@_axis_nan_policy_factory(
    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1
)
```
I think this will be sufficient.  If you go with the decorator, go ahead and remove all the `nan_policy` stuff (including separate use of `nanmean`, docstring, etc.) from your code. There may be a speed advantage to `nanmean`, but that is something we can optimize later. The decorator makes sure all the edge cases are right, as you mentioned.

Then we'll also want to add a test case like all these:
https://github.com/scipy/scipy/blob/63a3ed9df86a637144c8f378dcdeb5ab5b6ecdfd/scipy/stats/tests/test_axis_nan_policy.py#L31-L39